### ISSUE TITLE
DataWindow should respect the focusFirstControl parameter

### ISSUE THREAD

**Issue by MaximKopylov** [Catel/Catel#1259]
### Steps to reproduce

1. Create DataWindow:
```
<catel:DataWindow x:Class="SimpleTest.Views.MainWindow
                  xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                  xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                  xmlns:catel="http://schemas.catelproject.com"
                  ResizeMode="CanResize">
  <Grid>
    <TextBox></TextBox>
  </Grid>
</catel:DataWindow>
```
2. Modify constructor in code-behind:
```
public MainWindow() : base(DataWindowMode.Custom, focusFirstControl:false)
{
  InitializeComponent();
}
```

Platform:
.NET version: 4.5 and newer

### Expected behaviour

The TextBox should not have focus. Expected behavior like a normal window, when the TextBox has no focus when the window is displayed.

### Actual behaviour

The Text Box always have focus when window are shown.

**Comment by GeertvanHorrik** [comment]
Please prepare a small repro for us so we don't have to spend time on trying to reproduce this.

**Comment by MaximKopylov** [comment]
This is a simple example: https://github.com/MaximKopylov/CatelTest
In this example TextBox got a focus on startup.

**Comment by MaximKopylov** [comment]
What about the unexpected behaviour? Can you reproduce that? The bug is confirmed or not?

**Comment by GeertvanHorrik** [comment]
We will investigate when we have time. Lots of tickets, and supporting devs go first.

**Comment by GeertvanHorrik** [comment]
I think this could be the issue:

https://github.com/Catel/Catel/blob/develop/src/Catel.MVVM/Windows/Windows/DataWindow/DataWindow.cs#L844

**Comment by GeertvanHorrik** [comment]
Thanks for reporting, we have fixed it in 5.10 (alpha).
